### PR TITLE
Support unicode with line feed in CLI print 

### PR DESCRIPTION
**Pull request by lakshmi-kannan**
Fixes: https://github.com/StackStorm/st2/issues/3510

See https://mail.python.org/pipermail/python-list/2006-January/411909.html. 

TODO

- [x] Validate other prints (rule, sensor etc etc) actually use this util. 

Test cases without any code change:

```
======================================================================
ERROR: test_unicode_with_line_feed_like_chars (tests.unit.test_util_strutil.StrUtilTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/src/storm/st2/st2client/tests/unit/test_util_strutil.py", line 38, in test_unicode_with_line_feed_like_chars
    out_str = unescape(in_str)
  File "/mnt/src/storm/st2/st2client/st2client/utils/strutil.py", line 20, in unescape
    if isinstance(s, basestring) else s)
UnicodeEncodeError: 'ascii' codec can't encode character u'\u20ac' in position 0: ordinal not in range(128)
```

After the fix: 

```
test_unicode_with_line_feed_like_chars (tests.unit.test_util_strutil.StrUtilTestCase) ... ok
```

With the fix:

```
(virtualenv)vagrant@st2dev /m/s/s/st2 ❯❯❯ st2 action list --pack=cms                            issues/3510/unicode_support_action_list ✭ ✱
+------------------------+------+------------------------------+
| ref                    | pack | description                  |
+------------------------+------+------------------------------+
| cms.get_group_info     | cms  | 根据group_id查询group信息    |
| cms.get_pool_info      | cms  | 根据pool_id查询pool信息      |
| cms.get_server_info    | cms  | 根据pool_id查询server信息    |
| cms.get_site_info      | cms  | 根据group_id查询站点信息     |
| cms.get_vd_info        | cms  | 根据group_id查询虚拟目录信息 |
| cms.group_deletion     | cms  | 调用CMS接口删除Group         |
| cms.pool_deletion      | cms  | 调用CMS接口删除Group         |
| cms.site_deletion      | cms  | 调用CMS接口删除站点          |
| cms.ticket_association | cms  | 关联单子                     |
| cms.vd_deletion        | cms  | 调用CMS接口删除虚拟目录      |
+------------------------+------+------------------------------+
(virtualenv)vagrant@st2dev /m/s/s/st2 ❯❯❯
```

### CODE REVIEW

**Comment by Kami** [comment]
So far LGTM :+1:

**Comment by lakshmi-kannan** [comment]
Once someone approves, I'll add CHANGELOG and merge to master.

File: st2client/st2client/formatters/execution.py [hunk 1]
```diff
@@ -60,4 +60,4 @@ def format(cls, entry, *args, **kwargs):
                     value = ('\n' if isinstance(value, dict) else '') + formatted_value
                 output += ('\n' if output else '') + '%s: %s' % \
                     (DisplayColors.colorize(attr, DisplayColors.BLUE), value)
-        return strutil.unescape(output)
```
Comment (lakshmi-kannan) [inline st2client/st2client/formatters/execution.py]: When we print execution result as response to st2 run core.local cmd="echo '¯_(ツ)_/¯'", I noticed that the entire YAML formatted response (a string) like 
```
id: 59516ff0d9d7ed651f394c18
status: succeeded
parameters:
  cmd: "echo '\xAF_(\u30C4)_/\xAF'"
result:
  failed: false
  return_code: 0
  stderr: ''
  stdout: "\xAF_(\u30C4)_/\xAF"
  succeeded: true
```
was being unescaped. Any unescaping should be done on the strings prior to YAML formatting. So I moved it to before.

File: st2client/st2client/utils/strutil.py [hunk 2]
```diff
@@ -15,5 +15,5 @@
 
 
 def unescape(s):
-    return (s.replace('\\r\\n', '\\n').decode('unicode_escape').encode('utf-8')
+    return (s.replace('\r\n', '\n')
```
Comment (lakshmi-kannan) [inline st2client/st2client/utils/strutil.py]: No need to escape `\`. Python treats \r as a single character.

In [1]: a = '\r'

In [2]: len(a)
Out[2]: 1

In [3]:

**Comment by Kami** [comment]
Circle CI tests are failing, looks like some tests still need to be updated?

File: st2client/st2client/formatters/execution.py (hunk 1 above)
Comment (Kami) [inline st2client/st2client/formatters/execution.py]: Good catch :+1:

File: st2client/tests/unit/test_util_strutil.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    def test_strip_carriage_returns(self):
+        in_str = 'Windows editors introduce\r\nlike a noob in 2017.'
+        out_str = strutil.strip_carriage_returns(in_str)
+        exp_str = 'Windows editors introduce\nlike a noob in 2017.'
+        self.assertEqual(out_str, exp_str)
+
+    def test_unicode_with_line_feed_like_chars(self):
+        in_str = u'€\\n€'
+        out_str = strutil.unescape(in_str).encode('utf-8').decode('string_escape').decode('utf-8')
```
Comment (lakshmi-kannan) [inline st2client/tests/unit/test_util_strutil.py]: We just want to make sure that struil.enescape(in_str) should be equal to out_str, right?

Comment (m4dcoder) [inline st2client/tests/unit/test_util_strutil.py]: Actually, I'm not sure what this test is suppose to test... Maybe to test all the decode/encode previously in the code?

**Pull request merged**